src/rosdiscover/recover/model.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+__all__ = ("RecoveredNodeModel",)
+
+import json
+import typing as t
+
+import attr
+
+from ..interpreter import NodeContext
+
+
+@attr.s(frozen=True, slots=True)
+class RecoveredNodeModel:
+    """Provides a description of the run-time architecture of a given node.
```

Comment:
It looks like this class just stores information about the pre-analyzed node, rather than information about the pubs, subs, etc. that is the run-time architecture. So, I think the description is right, but the fields are wrong?

Reply:
The fields should be okay. The description is a little misleading, though. The description makes it sound like we're providing the concrete node model for a particular configuration, whereas what we're actually doing is providing a symbolic program that can be used to obtain a concrete set of pubs, subs, etc. using the interpreter.

Reply:
Improved description to eliminate ambiguity